crates/matrix-sdk-crypto/src/session_manager/group_sessions/share_strategy.rs

```diff
@@ -401,6 +403,61 @@ fn split_devices_for_user(
     device_owner_identity: &Option<UserIdentityData>,
     only_allow_trusted_devices: bool,
     error_on_verified_user_problem: bool,
+) -> DeviceBasedRecipientDevices {
+    if only_allow_trusted_devices {
+        let recipient_devices = split_devices_for_user_for_only_trusted_devices(
+            user_devices,
+            own_identity,
+            device_owner_identity,
+        );
+        DeviceBasedRecipientDevices::Devices(recipient_devices)
+    } else if error_on_verified_user_problem {
+        split_devices_for_user_for_error_on_verified_user_problem_strategy(
+            user_devices,
+            own_identity,
+            device_owner_identity,
+        )
```

Comment:
it's worth noting that we never tested what happened if you set both `error_on_verified_user_problem` and `only_allow_trusted_devices`, and I don't think the behaviour you actually got made very much sense. You actually ended up with behaviour which was *weaker* than ` error_on_verified_user_problem` on its own.

In particular, it meant we sent a withheld message to unverified devices belonging to verified users, whereas with `error_on_verified_user_problem` alone, that would result in an error being returned (and no messages being sent). 

In this PR, I'm giving precedence to `only_allow_trusted_devices` as, other than in that case, it is much the stricter setting.